### ISSUE TITLE
_TestingModelBackend incorrectly raises KeyError on missing relation

### ISSUE THREAD

**Issue by gnuoy** [canonical/operator#241]
relation_ids from _TestingModelBackend  should return an empty list if there is no relation matching the supplied realtion_id (as per the behaviour of the ModelBackend *1) instead of throwing a KeyError.
To reproduce:
```
from ops.testing import Harness
from ops.charm import CharmBase

harness = Harness(
    CharmBase,
    meta='''
        provides:
          some-relation:
            interface: some-interface
    ''')
harness.begin()
harness.charm.model.get_relation('some-relation')
```
*1 https://github.com/canonical/operator/blob/master/ops/model.py#L699

**Comment by gnuoy** [comment]
I think https://github.com/canonical/operator/pull/196 will resolve this

**Comment by chipaca** [comment]
@gnuoy I think so too. Can you try it and let us know?

**Comment by gnuoy** [comment]
@chipaca it does fix this issue, I've just put a comment on #196 saying as much.
